check-log/lib/check-log.go

```diff
@@ -431,6 +415,39 @@ func (opts *logOpts) match(line string) (bool, []string) {
 	return true, matches
 }
 
+func parseFilePattern(directory string, filePattern string, caseInsensitive bool) ([]string, error) {
+	var dirStr string
+	var filePat string
+	if directory != "" {
+		dirStr = directory
+		filePat = filePattern
+	} else {
+		dirStr = filepath.Dir(filePattern)
+		filePat = filepath.Base(filePattern)
+	}
```

Comment:
We should add a comment that these lines is provided for backward compatibility.